[network-key] change name from `master` to `network`

jwhui opened the pull request:
To conform to [SPEC-982](https://threadgroup.atlassian.net/jira/core/projects/SPEC/issues/SPEC-982).

Depends-On: openthread/ot-br-posix#889
Depends-On: openthread/pyspinel#116
Depends-On: openthread/ot-ns#132

simonlingoogle commented on tools/harness-thci/OpenThread_WpanCtl.py:
```diff
@@ -2347,8 +2347,8 @@ def MGMT_ACTIVE_SET(
 
             if xMasterKey is not None:
```
Should we fix `xMasterKey` too?

jwhui replied:
Is `MGMT_ACTIVE_SET()` called by test harness? Will updating `xMasterKey` affect compatibility with test harness? If not, then we should update this as well.

simonlingoogle commented:
Found `master` in these files:
- examples/platforms/simulation/README.md
- script/check-api-version
- tests/scripts/thread-cert/
- tools/harness-thci

@jwhui

jwhui commented:
> Found `master` in these files:
> 
> * examples/platforms/simulation/README.md

Fixed.

> * script/check-api-version

Fixed.

> * tests/scripts/thread-cert/

Changing `thread_meshcop.tlv.master_key` breaks compatibility with dissector.

> * tools/harness-thci

Will changing `xMasterKey` break compatibility with test harness?

abtink commented on src/core/meshcop/dataset.hpp:
```diff
@@ -857,13 +857,13 @@ class Dataset
      *
      * @param[in]  aInstance           A reference to the OpenThread instance.
-     * @param[out] aIsMasterKeyUpdated A pointer to where to place whether master key was updated.
+     * @param[out] aIsNetworkKeyUpdated A pointer to where to place whether network key was updated.
```
nit: align with `aInstance`

abtink reviewed (approved):
Looks great! Thanks for doing this.
Just found one small alignment in comment.

simonlingoogle commented:
> > * tests/scripts/thread-cert/
> 
> Changing `thread_meshcop.tlv.master_key` breaks compatibility with dissector.

Yes. We can't fix it for now.


> > * tools/harness-thci
> 
> Will changing `xMasterKey` break compatibility with test harness?

It will. We can't fix it for now.

simonlingoogle reviewed (approved):
LGTM :+1:
